processor/ratelimitprocessor/metadata.yaml

```diff
@@ -10,3 +10,31 @@ tests:
   config:
     rate: 1
     burst: 1
+
+telemetry:
+  metrics:
+    ratelimit.requests:
+      enabled: true
+      description: Number of rate-limiting requests
+      unit: "{requests}"
+      sum:
+        value_type: int
+        monotonic: true
+      attributes: ["project_id", "processor_id", "ratelimit_decision", "limit_threshold", "reason"]
+
+attributes:
```

Comment:
Please see this comment https://github.com/elastic/opentelemetry-collector-components/pull/562#discussion_r2071880991, We dont want to expose the `project_id` etc details, instead keep this to attributes being derived from metadata_keys. Not sure if thats something possible with the metadata schema, but we can come back to later as well.

Reply:
Ahh sorry I missed it. Will remove `project_id`. Yeah I dont think the current `mdatagen` supports dynamic attribute keys. Can look into it more later.
How about `processor_id` then? ProcessorID is from `telemetry.WithProcessorID(r.set.ID.String())`.